### DISCUSSION TITLE
Need help: ST7735 TFT screen cannot readpixel

### DISCUSSION THREAD

**Discussion by GuoWQ222** [Bodmer/TFT_eSPI#2832] [Q&A - General]
Hi, everyone. 
When I try to run the **TFT_ReadWrite_Test.ino** on my esp32-wroom-32 board, the ST7735 screen doesn't seem to readpixel but always returns 0 instead.
The 128*160 screen is using the SDA line. It only has GND,VDD,SCL,SDA,RST,DC,CS,BLK pins.
**So I wonder if my screen support reading data via the SDA line.**

Here's my **User_Setup.h**:
#define USER_SETUP_INFO "User_Setup"

#define ST7735_DRIVER      // Define additional parameters below for this display

 #define TFT_SDA_READ      // This option is for ESP32 ONLY, tested with ST7789 and GC9A01 display only

  #define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue

 #define TFT_WIDTH  128

 #define TFT_HEIGHT 160

 #define ST7735_GREENTAB2

#define TFT_MOSI 15 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 14
#define TFT_CS   5  // Chip select control pin
#define TFT_DC   27  // Data Command control pin
#define TFT_RST  33  // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   22  // LED back-light

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  27000000

#define SPI_READ_FREQUENCY  20000000

#define SPI_TOUCH_FREQUENCY  2500000

**I really appreciate your help.**

**Comment by Bodmer** [comment]
This indicates the display does not support bi-directional transfer. Some boards do not provide that capability.
